src/v/compression/tests/lz4_buf_tests.cc

```diff
@@ -96,3 +98,23 @@ TEST(MixedAllocations, PassThroughAlloc) {
     allocator.customFree(allocator.opaqueState, random_alloc);
     EXPECT_EQ(buffers.state, no_buffers_allocated);
 }
+
+TEST(CustomAllocDisabled, Configuration) {
+    compression::reset_lz4_decompression_buffers();
```

Comment:
Might not be a problem, but consider adding a test fixture that calls this in the setup and teardown methods. If other tests use the thread locals but forget to clear them, it may result in pretty surprising test results

Reply:
added a fixture